spk/salt-minion/src/service-setup.sh

```diff
@@ -0,0 +1,35 @@
+PYTHON_DIR="/usr/local/python3"
+SALT_PKG_DIR="/usr/local/salt-minion"
+PATH="${SYNOPKG_PKGDEST}/bin:${SYNOPKG_PKGDEST}/env/bin:${PYTHON_DIR}/bin:${PATH}"
+VIRTUALENV="${PYTHON_DIR}/bin/virtualenv"
+PYTHON="${SALT_PKG_DIR=}/env/bin/python"
+LANGUAGE="env LANG=en_US.UTF-8"
+SALT_MINION="${SALT_PKG_DIR}/env/bin/salt-minion"
+SALT_CONF_DIR="/var/salt/etc"
```

Comment:
Is there a reason why to prefer to move configuration out of package?
To remain in SynoCommunity package policy, I would recommend
```suggestion
SALT_CONF_DIR="${SYNOPKG_PKGDEST}/var/"
```
so that configuration is read at `${SYNOPKG_PKGDEST}/var/minion.conf` including parts as `${SYNOPKG_PKGDEST}/var/minion.d/*.conf`
What do you think about it?

Reply:
I just sent a commit moving config back to ${SYNOPKG_PKGDEST}/var, but I still had to define a variable SALT_PKG_DIR same as  SYNOPKG_PKGDEST, because SYNOPKG_PKGDEST seems to be defined only at install time.
If I use  SYNOPKG_PKGDEST to define PID_FILE and config location in SERVICE_COMMAND, service doesn't work because SYNOPKG_PKGDEST is empty when the script start-stop-status runs. I don't know if it's a bug or a feature (or I'm doing something wrong)

Reply:
In fact you have to use `synopkg` command to operate DSM 6 service: https://github.com/SynoCommunity/spksrc/wiki/Frequently-Asked-Questions#how-to-query-package-status-or-start-it-from-command-line

Reply:
Sorry for not finding that FAQ
I was using synopkg tool, but status gave me an error and trying to diagnose I run the script directly and found that "problem"

Maybe you can point me to where I can find info to fix the status command in synopkg. 

If I run script directly it works
```
root@ctkbackups:~# SYNOPKG_PKGNAME="salt-minion"
root@ctkbackups:~# SYNOPKG_PKGDEST=/var/packages/${SYNOPKG_PKGNAME}/target
root@ctkbackups:~# SYNOPKG_DSM_VERSION_MAJOR=6
root@ctkbackups:~# export SYNOPKG_PKGNAME SYNOPKG_PKGDEST SYNOPKG_DSM_VERSION_MAJOR
root@ctkbackups:~# /var/packages/${SYNOPKG_PKGNAME}/scripts/start-stop-status status
salt-minion is running
```

But if I use synopkg, it's not working ok (just the status command, start and stop work flawlessly)
```
root@ctkbackups:~# synopkg status salt-minion
salt-minion package status is stopped but unit is active, it is possible that package encounters an error after it was started successfully
Status: [310]
```

BTW, I also pushed a commit going back to SYNOPKG_PKGDEST in start-stop-status script

Reply:
"status" action is looking for process existence based on PID number visible in pid file. https://github.com/SynoCommunity/spksrc/blob/master/mk/spksrc.service.start-stop-status#L79
You have set `pidfile` option in configuration file, `PID_FILE` variable for framework is consistent. So I propose you check manually content of pid file to investigate why "status" does not report proper value.

Reply:
Ok, I found the problem:
Using the start-stop-script in cli worked, using "initctl status" worked, but using "synopkg status" didn't work
I used strace from Synogear and saw that there was an ACCEES DENIED error when trying to read pid file, si it seemed synopkg was not running as root.
I checked my privileges file (wich I copied from wiki) and noticed there was no entry for "status" action
I added the status action to be run as root, and now synopkg status works

Maybe the sample privileges file should contain also the status action (https://github.com/SynoCommunity/spksrc/wiki/Service-Support#run-service-process-as-root)